include/secp256k1_silentpayments.h

```diff
@@ -25,6 +26,97 @@ extern "C" {
  *  any further elliptic-curve operations from the wallet.
  */
 
+/** This struct serves as an input parameter for passing the silent payment
+ *  address data to `silentpayments_sender_create_outputs`.
+ *
+ *  Index must be set to the position (starting with 0) of this recipient in the
+ *  `recipients` array passed to `silentpayments_sender_create_outputs`. It is
+ *  used to map the returned generated outputs back to the original recipient.
+ *
+ *  The spend public key field is named `labeled_spend_pubkey` to indicate this
+ *  spend public key may be tweaked with an optional label. This is not relevant
+ *  for the sender and is purely a documentation convention to differentiate
+ *  between other uses of `spend_pubkey` in this API, where it is meant to refer
+ *  to the unlabeled spend public key.
+ */
+typedef struct secp256k1_silentpayments_recipient {
+    secp256k1_pubkey scan_pubkey;
+    secp256k1_pubkey labeled_spend_pubkey;
+    size_t index;
+} secp256k1_silentpayments_recipient;
+
+/** Create Silent Payment outputs for recipient(s).
+ *
+ *  Given a list of n secret keys a_1...a_n (one for each silent payment
+ *  eligible input to spend), a serialized outpoint, and a list of recipients,
+ *  create the taproot outputs. Inputs with conditional branches or multiple
+ *  public keys are excluded from silent payments eligible inputs; see BIP352
+ *  for more information.
+ *
+ *  `outpoint_smallest` refers to the smallest outpoint lexicographically
+ *  from the transaction inputs (both silent payments eligible and non-eligible
+ *  inputs). This value MUST be the smallest outpoint out of all of the
+ *  transaction inputs, otherwise the recipient will be unable to find the
+ *  payment. Determining the smallest outpoint from the list of transaction
+ *  inputs is the responsibility of the caller. It is strongly recommended
+ *  that implementations ensure they are doing this correctly by using the
+ *  test vectors from BIP352.
+ *
+ *  If necessary, the taproot secret keys are negated to enforce even y-parity.
+ *  For that reason, the secret keys have to be passed in via two different
+ *  parameter pairs, depending on whether the secret keys correspond to x-only
+ *  (taproot) outputs or not.
+ *
+ *  Returns: 1 if creation of outputs was successful.
+ *           0 if the input private keys sum to zero,
+ *             or input_hash or hash(shared_secret || k) are invalid scalars
+ *             (stastically improbable).
```

Comment:
```suggestion
 *             (statistically improbable).
```
(here and in many other instances)

Reply:
Turns out I am not as good a typist as I thought I was 😅